sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParser.java

```diff
@@ -117,108 +118,110 @@ private void handleSegmentPointTag(XmlParserHelper xmlParserHelper) {
+    // The key is the line number.
+    // The value is the aggregated coverage for each sequence point on the line.
+    private Map<Integer, SequencePointsCoverage> sequencePointCoveragePerLine;
 
-    LineComplexCoverage(String fileId) {
-      this.fileId = fileId;
-      this.methodHitsPerLine = new HashMap<>();
+    DetailedLineCoverage(String filePath) {
+      this.filePath = filePath;
+      this.sequencePointCoveragePerLine = new HashMap<>();
     }
 
-    void addCoverageForLineAndMethod(Integer lineId, String methodName, int hits) {
-      methodHitsPerLine.putIfAbsent(lineId, new MethodHits());
-      MethodHits methodHits = methodHitsPerLine.get(lineId);
-      methodHits.addHit(methodName, hits);
+    /**
+     * @param lineId - line number
+     * @param sequencePointIdentifier - identifier for the sequence point - must be unique inside the file
+     * @param hits - number of hits (visit count)
+     */
+    void addCoverageForLineAndSequencePoint(Integer lineId, String sequencePointIdentifier, int hits) {
+      sequencePointCoveragePerLine.putIfAbsent(lineId, new SequencePointsCoverage());
+      SequencePointsCoverage sequencePointsCoverage = sequencePointCoveragePerLine.get(lineId);
+      sequencePointsCoverage.addHit(sequencePointIdentifier, hits);
     }
 
     /**
      * Transfers the coverage data to the Coverage object.
      * @param coverage - the object which centralizes coverage information.
      */
     void transferData(Coverage coverage) {
-      LOG.trace("Found coverage information about '{}' lines having single-line method declarations for file '{}'",
-        methodHitsPerLine.size(), fileId);
-
-      for (Map.Entry<Integer, MethodHits> lineMethodHits : methodHitsPerLine.entrySet()) {
-
-        Integer lineId = lineMethodHits.getKey();
-        MethodHits methodHits = lineMethodHits.getValue();
-        // where we have both covered and uncovered, use branch coverage
-        // e.g. auto-implemented property with getter and setter
-        if (methodHits.size() > 1) {
-          int coveredMethods = methodHits.countMethodsWithCoverage();
-          // this is not really branch coverage, but it's better than nothing
-          coverage.addBranchCoverage(fileId, new BranchCoverage(lineId, methodHits.size(), coveredMethods));
+      LOG.trace("Found coverage information about '{}' lines having multiple sequence points for file '{}'",
+        sequencePointCoveragePerLine.size(), filePath);
+
+      for (Map.Entry<Integer, SequencePointsCoverage> lineSequencePointCoverage : sequencePointCoveragePerLine.entrySet()) {
+
+        Integer lineId = lineSequencePointCoverage.getKey();
+        SequencePointsCoverage sequencePointsCoverage = lineSequencePointCoverage.getValue();
+
+        // where there are multiple sequence points on a line, use branch coverage
+        if (sequencePointsCoverage.size() > 1) {
+          int coveredSequencePointsOnLine = (int) sequencePointsCoverage.countCoveredSequencePoints();
```

Comment:
I don't think it's likely to have sequence points with more that Int.MAX_VALUE hits

but maybe I should do a test to fail nicely?

Reply:
or just change the value in `Coverage` to use long, unless the SQ API requires int

Reply:
I will just leave it as is